packages/runtime/test/sparql.test.ts

```diff
@@ -0,0 +1,156 @@
+_:b33 <http://purl.org/spar/fabio/hasManifestation> _:b31 .
+_:b33 <http://purl.org/spar/fabio/hasManifestation> _:b32 .
+_:b34 <http://purl.org/spar/pro/isDocumentContextFor> _:b27 .
+_:b34 <http://purl.org/spar/pro/isDocumentContextFor> _:b29 .
+_:b34 <http://purl.org/spar/pwo/produces> _:b33 .
+_:b35 <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://purl.org/spar/fabio/Preprint> .
+_:b35 <http://prismstandard.org/namespaces/basic/2.0/doi> "10.1101/2022.11.08.515698" .
+_:b36 <http://purl.org/spar/pso/isStatusHeldBy> _:b35 .
+_:b36 <http://purl.org/spar/pso/withStatus> <http://purl.org/spar/pso/peer-reviewed> .
+_:b37 <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://purl.org/spar/fabio/Preprint> .
+_:b37 <http://prismstandard.org/namespaces/basic/2.0/doi> "10.1101/2022.11.08.515698" .
+_:b37 <http://purl.org/spar/fabio/hasURL> "https://www.biorxiv.org/content/10.1101/2022.11.08.515698v2"^^<http://www.w3.org/2001/XMLSchema#anyURI> .
+_:b2 <http://www.ontologydesignpatterns.org/cp/owl/taskexecution.owl#isExecutedIn> _:b18 .
+_:b2 <http://www.ontologydesignpatterns.org/cp/owl/taskexecution.owl#isExecutedIn> _:b25 .
+_:b2 <http://www.ontologydesignpatterns.org/cp/owl/taskexecution.owl#isExecutedIn> _:b34 .
+_:b2 <http://purl.org/spar/pso/resultsInAcquiring> _:b36 .
+_:b2 <http://purl.org/spar/pwo/needs> _:b37 .
+_:b2 <http://purl.org/spar/pwo/hasPreviousStep> _:b1 .
+<https://data-hub-api.elifesciences.org/enhanced-preprints/docmaps/v1/get-by-doi?preprint_doi=10.1101%2F2022.11.08.515698> <http://purl.org/spar/pwo/hasStep> _:b0 .
+<https://data-hub-api.elifesciences.org/enhanced-preprints/docmaps/v1/get-by-doi?preprint_doi=10.1101%2F2022.11.08.515698> <http://purl.org/spar/pwo/hasStep> _:b1 .
+<https://data-hub-api.elifesciences.org/enhanced-preprints/docmaps/v1/get-by-doi?preprint_doi=10.1101%2F2022.11.08.515698> <http://purl.org/spar/pwo/hasStep> _:b2 .
+<https://data-hub-api.elifesciences.org/enhanced-preprints/docmaps/v1/get-by-doi?preprint_doi=10.1101%2F2022.11.08.515698> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://purl.org/spar/pwo/Workflow> .
+`
+
+test('constructs and extracts a realistic docmap from a sparql backend', async (t) => {
+  const backend = new OxigraphInmemBackend()
+  const graph = oxigraph.defaultGraph()
+  backend.store.load(
+    data, // data
+    'application/n-triples', // mime type
+    'https://docmaps-project.github.io/ex/v1/', // base
+    graph,
+  )
+
+  const processor = new SparqlAdapter(backend)
+
+  const docmap = await processor.docmapWithIri(
+    'https://data-hub-api.elifesciences.org/enhanced-preprints/docmaps/v1/get-by-doi?preprint_doi=10.1101%2F2022.11.08.515698',
+  )()
+
+  if (E.isLeft(docmap)) {
+    t.fail(`got error instead of docmap: ${inspect(docmap.left, { depth: 4 })}`)
+    return
+  }
+
+  t.is(docmap.right.id, "https://data-hub-api.elifesciences.org/enhanced-preprints/docmaps/v1/get-by-doi?preprint_doi=10.1101%2F2022.11.08.515698")
```

Comment:
The main test assertion here is that by whatever means we find appropriate triples in the datastore, we should be able to extract them and serialize using jsonld to a decodable Docmap format. Excitingly, this does work based on the work of last week getting the SDK to handle tripes-to-jsonld serialization.